File: postman/sdks/Collections SDKs/iconocracy-corpus/cli/sdk/iconocracycorpussdk.go
package iconocracycorpussdk

import (
	"example.com/iconocracy-corpus-sdk/sdk/get"
	"example.com/iconocracy-corpus-sdk/sdk/implementedapi"
	"example.com/iconocracy-corpus-sdk/sdk/internal/clients/rest/hooks"
	"example.com/iconocracy-corpus-sdk/sdk/internal/configmanager"
	"example.com/iconocracy-corpus-sdk/sdk/post"
	"time"
)

// IconocracyCorpusSDK is the main SDK client that provides access to all service endpoints.
// It manages configuration, authentication, and service instances with centralized settings.
type IconocracyCorpusSDK struct {
	Get            *get.Service
	ImplementedAPI *implementedapi.Service
	Post           *post.Service
	manager        *configmanager.ConfigManager
}

func NewIconocracyCorpusSDK(config Config) *IconocracyCorpusSDK {
	get := get.NewService()
	implementedAPI := implementedapi.NewService()
	post := post.NewService()

	manager := configmanager.NewConfigManager(config)
	hook := hooks.NewDefaultHook()
	get.WithConfigManager(manager)
	implementedAPI.WithConfigManager(manager)
	post.WithConfigManager(manager)
	get.WithHook(hook)
	implementedAPI.WithHook(hook)
	post.WithHook(hook)

	return &IconocracyCorpusSDK{
		Get:            get,
		ImplementedAPI: implementedAPI,
		Post:           post,
		manager:        manager,
	}
}

func (i *IconocracyCorpusSDK) SetBaseURL(baseURL string) {
	i.manager.SetBaseURL(baseURL)
}

func (i *IconocracyCorpusSDK) SetTimeout(timeout time.Duration) {
	i.manager.SetTimeout(timeout)
}

// SetEnvironment configures the SDK to use the specified environment's base URL.
func (i *IconocracyCorpusSDK) SetEnvironment(environment Environment) {
	i.manager.SetBaseURL(string(environment))
}

// c029837e0e474b76bc487506e8799df5e3335891efe4fb02bda7a1441840310c

File: postman/sdks/Collections SDKs/iconocracy-corpus/java/src/main/java/com/iconocracycorpussdk/IconocracyCorpusSdk.java
package com.iconocracycorpussdk;

import com.iconocracycorpussdk.config.IconocracyCorpusSdkConfig;
import com.iconocracycorpussdk.http.Environment;
import com.iconocracycorpussdk.http.interceptors.DefaultHeadersInterceptor;
import com.iconocracycorpussdk.http.interceptors.RetryInterceptor;
import com.iconocracycorpussdk.services.GetService;
import com.iconocracycorpussdk.services.ImplementedApiService;
import com.iconocracycorpussdk.services.PostService;
import java.util.concurrent.TimeUnit;
import okhttp3.OkHttpClient;

/**
 * Main SDK client class for IconocracyCorpusSdk.
 * Provides centralized access to all service endpoints with configurable authentication,
 * environment management, hooks, and HTTP client settings.
 */
public class IconocracyCorpusSdk {

  public final GetService get;
  public final ImplementedApiService implementedApi;
  public final PostService post;

  private final IconocracyCorpusSdkConfig config;

  /**
   * Constructs a new instance of IconocracyCorpusSdk with default configuration.
   */
  public IconocracyCorpusSdk() {
    // Default configs
    this(IconocracyCorpusSdkConfig.builder().build());
  }

  /**
   * Constructs a new instance of IconocracyCorpusSdk with custom configuration.
   * Initializes all services, HTTP client, and optional OAuth token manager.
   *
   * @param config The SDK configuration including base URL, authentication, timeout, and retry settings
   */
  public IconocracyCorpusSdk(IconocracyCorpusSdkConfig config) {
    this.config = config;

    final OkHttpClient httpClient = new OkHttpClient.Builder()
      .addInterceptor(new DefaultHeadersInterceptor(config))
      .addInterceptor(new RetryInterceptor(config.getRetryConfig()))
      .readTimeout(config.getTimeout(), TimeUnit.MILLISECONDS)
      .build();

    this.get = new GetService(httpClient, config);
    this.implementedApi = new ImplementedApiService(httpClient, config);
    this.post = new PostService(httpClient, config);
  }

  /**
   * Sets the environment for all API requests.
   *
   * @param environment The environment to use (e.g., DEFAULT, PRODUCTION, STAGING)
   */
  public void setEnvironment(Environment environment) {
    setBaseUrl(environment.getUrl());
  }

  /**
   * Sets the base URL for all API requests.
   *
   * @param baseUrl The base URL to use for API requests
   */
  public void setBaseUrl(String baseUrl) {
    this.config.setBaseUrl(baseUrl);
  }
}
// c029837e0e474b76bc487506e8799df5e3335891efe4fb02bda7a1441840310c
